src/message_filter.rs

```diff
@@ -29,7 +30,11 @@ fn hash<T: Hash>(t: &T) -> u64 {
 /// A time based message filter that takes any generic type as a key and will drop keys after a
 /// time period (LRU Cache pattern).
 pub struct MessageFilter<Message> {
-    entries: Vec<TimestampedMessage>,
+    /// The number of times each message has been received so far, and the expiry timestamp.
+    count: HashMap<u64, (usize, Instant)>,
```

Comment:
Because `HashMap` is faster and we don't iterate over it, so the order should never matter in this case.